op-batcher/compressor/blind_compressor.go

```diff
@@ -0,0 +1,77 @@
+package compressor
+
+import (
+	"bytes"
+	"compress/zlib"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+)
+
+// BlindCompressor is a simple compressor that blindly compresses data
+// the only way to know if the target size has been reached is to first flush the buffer
+// and then check the length of the compressed data
+type BlindCompressor struct {
+	config Config
+
+	inputBytes int
+	buf        bytes.Buffer
+	compress   *zlib.Writer
+}
+
+// NewBlindCompressor creates a new derive.Compressor implementation that compresses
+func NewBlindCompressor(config Config) (derive.Compressor, error) {
+	c := &BlindCompressor{
+		config: config,
+	}
+
+	compress, err := zlib.NewWriterLevel(&c.buf, zlib.BestCompression)
```

Comment:
The largest challenge to that right now is that to access `TargetOutputSize`, the Compressor interface itself must expose it. Because the Compressor Interface lives in derive, but the compressors themselves live in `compressor`, trying to get this value causes import cycles.

We should refactor the dependency tree generally, but that's a bigger task than this PR.